src/linux_mcp_server/tools/storage.py

```diff
@@ -87,273 +109,161 @@ async def list_block_devices(
-                return f"Error: Path is not a directory: {path}"
-
-            if not os.access(path_obj, os.R_OK):
-                return f"Error: Permission denied to read directory: {path}"
-
-            path_str = str(path_obj)
-        else:
-            # For remote execution, use the path as-is
-            path_str = path
-
-        # Use du command to get directory sizes efficiently
-        returncode, stdout, _ = await execute_command(
-            ["du", "-b", "--max-depth=1", path_str],
-            host=host,
-            username=username,
-        )
-
-        # Parse output - du may return non-zero on permission errors but still give valid data
-        lines = stdout.strip().split("\n")
-        dir_sizes = []
-
-        for line in lines:
-            if not line:
-                continue
-            parts = line.split("\t", 1)
-            if len(parts) == 2:
-                try:
-                    size = int(parts[0])
-                    dir_path_str = parts[1]
-                    # Skip the parent directory itself
-                    dir_name = Path(dir_path_str).name
-                    if dir_path_str != path_str:
-                        dir_sizes.append((dir_name, size))
-                except (ValueError, IndexError):
-                    continue
-
-        if not dir_sizes:
-            # Only error if we got no output AND a bad return code
-            if returncode != 0:
-                return "Error: du command failed and returned no directory data"
-            return f"No subdirectories found in: {path}"
-
-        # Sort by size (descending) and take top N
-        dir_sizes.sort(key=lambda x: x[1], reverse=True)
-        top_dirs = dir_sizes[:validated_top_n]
-
-        # Format output
-        result = []
-        result.append(f"=== Top {len(top_dirs)} Largest Directories ===")
-        result.append(f"Path: {path_str}")
-        result.append(f"\nTotal subdirectories found: {len(dir_sizes)}\n")
-
-        for i, (dir_name, size) in enumerate(top_dirs, 1):
-            result.append(f"{i}. {dir_name}")
-            result.append(f"   Size: {format_bytes(size)}")
-
-        return "\n".join(result)
-
-    except Exception as e:
-        return f"Error analyzing directories: {str(e)}"
-
-
-@mcp.tool(
-    title="List directories by name",
-    description="List directories under a specified path sorted by name.",
-    annotations=ToolAnnotations(readOnlyHint=True),
-)
-@log_tool_call
-async def list_directories_by_name(
-    path: t.Annotated[str, "The directory path to analyze"],
-    reverse: t.Annotated[bool, "If True, sort in reverse alphabetical order (Z-A)"] = False,
-    host: Host | None = None,
-    username: Username | None = None,
-) -> str:
-    """
-    List directories under a specified path sorted by name.
-
-    This function uses efficient Linux primitives (find and sort) to list directories.
-    """
-    import os
-
+) -> t.Annotated[
+    Sequence[DirectoryEntry],
```

Comment:
I'm not sure Sequence is correct. The function returns a list which is modified after instantiation. A Sequence, while mutable, is meant to be read-only.

This is probably the reason the type hint `directories_by_size: list[DirectoryEntry]` is necessary. This can be avoided by not modifying the list or changing the return type.